akka-http-core/src/main/scala/akka/http/impl/engine/rendering/RenderSupport.scala

```diff
@@ -79,13 +79,13 @@ private object RenderSupport {
     override def onPush(elem: ByteString, ctx: Context[ByteString]): SyncDirective = {
       sent += elem.length
       if (sent > length)
-        throw InvalidContentLengthException(s"HTTP message had declared Content-Length $length but entity data stream amounts to more bytes")
+        ctx fail InvalidContentLengthException(s"HTTP message had declared Content-Length $length but entity data stream amounts to more bytes")
```

Comment:
I'm likely missing something here: I thought that throwing is equivalent to failStage, which I'd have thought would do the same as ctx.fail? 
`ctx.fail` seems cleaner in any case but wanted to see what I'm missing here.